tests/shared/src/test/scala/cats/tests/ParallelSuite.scala

```diff
@@ -390,6 +390,38 @@ class ParallelSuite
     }
   }
 
+  test("ParFlatMapN over List should be consistent with parMapN flatten") {
+    forAll { (as: List[Int], bs: List[Int], cs: List[Int]) =>
+      val mf: (Int, Int, Int) => List[Int] = _ :: _ :: _ :: Nil
+      assert((as, bs, cs).parFlatMapN(mf) == (as, bs, cs).parMapN(mf).flatten)
+    }
+  }
+
+  test("ParFlatMapN over NonEmptyList should be consistent with parMapN flatten") {
+    forAll { (as: NonEmptyList[Int], bs: NonEmptyList[Int], cs: NonEmptyList[Int]) =>
+      val mf: (Int, Int, Int) => NonEmptyList[Int] = (a, b, c) => NonEmptyList.of(a, b, c)
+      assert((as, bs, cs).parFlatMapN(mf) == (as, bs, cs).parMapN(mf).flatten)
+    }
+  }
+
+  test("ParFlatMap over List should be consistent with flatmap") {
+    forAll { as: List[Int] => assert(Tuple1(as).parFlatMap(_ :: Nil) == Tuple1(as).flatMap(_ :: Nil)) }
+  }
+
+  test("ParFlatMap over NonEmptyList should be consistent with flatmap") {
+    forAll { as: NonEmptyList[Int] =>
+      assert(Tuple1(as).parFlatMap(NonEmptyList.one) == Tuple1(as).flatMap(NonEmptyList.one))
+    }
+  }
+
+  test("ParMapN over f should be consistent with parFlatMapN over f lifted in Kleisli") {
+    forAll { (as: List[Int], bs: List[Int]) =>
+      val f: (Int, Int) => Int = (a, b) => a + b
+      val mf: (Int, Int) => List[Int] = (a, b) => List(a + b)
+      assert((as, bs).parMapN(f) == (as, bs).parFlatMapN(mf))
+    }
```

Comment:
What if we make `f` and `fm` more generic here?
```suggestion
    forAll { (as: List[Int], bs: List[Int], f: (Int, Int) => Int) =>
      val mf: (Int, Int) => List[Int] = (a, b) => List(f(a, b))
      assert((as, bs).parMapN(f) == (as, bs).parFlatMapN(mf))
    }
```
I mean that `f` does not have to be just a sum, does it?

Reply:
Wow, can scalacheck even generate random functions? I didn't knew that.

Reply:
I just sent a pr using it, and I fixed the test name

Reply:
HT @satorg @johnynek @armanbilge, this is a friendly reminder of this PR :)